libcudacxx/include/cuda/std/__exception/cuda_error.h

```diff
@@ -27,41 +27,53 @@
 #endif // !_CCCL_CUDA_COMPILER_NVCC && !_CCCL_CUDA_COMPILER_NVHPC
 
 #include <cuda/std/__exception/terminate.h>
-#include <cuda/std/detail/libcxx/include/stdexcept>
 
-#include <nv/target>
 #if !defined(_CCCL_COMPILER_NVRTC)
 #  include <cstdio>
+#  include <stdexcept>
 #endif // !_CCCL_COMPILER_NVRTC
 
+#include <nv/target>
+
 _LIBCUDACXX_BEGIN_NAMESPACE_CUDA
 
 /**
  * @brief Exception thrown when a CUDA error is encountered.
  */
-class cuda_error : public _CUDA_VSTD_NOVERSION::runtime_error
+
+#ifndef _LIBCUDACXX_NO_EXCEPTIONS
+class cuda_error : public ::std::runtime_error
 {
 public:
   cuda_error(::cudaError_t __status, const char* __msg) noexcept
-      : _CUDA_VSTD_NOVERSION::runtime_error("")
+      : ::std::runtime_error("")
   {
-    ::snprintf(
-      const_cast<char*>(this->what()), _CUDA_VSTD::__libcpp_refstring::__length, "cudaError %d: %s", __status, __msg);
+    char __formatted_msg[256];
+    ::snprintf(__formatted_msg, 256, "cudaError %d: %s", __status, __msg);
+
+    ::std::runtime_error __err{__formatted_msg};
+    *static_cast<::std::runtime_error*>(this) = __err;
```

Comment:
Suggestion: Using the assignment operator in the construct strikes me as odd. I would have used a separate function to format the message and then call it in the initialization list:
```suggestion
private:
  auto format(::cudaError_t __status, const char* __msg) -> std::string {
    char __formatted_msg[256];
    ::snprintf(__formatted_msg, 256, "cudaError %d: %s", __status, __msg);
    return msg;
  }
  
public:
  cuda_error(::cudaError_t __status, const char* __msg) noexcept
      : ::std::runtime_error(format(__status, __msg) {}
```
The version in this PR could be better, if `std::runtime_error` does not store an instance of `std::string` though.